python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -236,6 +241,29 @@ def to_default_asset_events(
                         metadata=metadata,
                     )
 
+    def _process_adapter_response_metadata(
+        self, adapter_response: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Process the adapter response metadata for a dbt CLI event.
+
+        The main interface for AdapterResponse is found at https://github.com/dbt-labs/dbt-adapters.
+
+        Currently, we pre-process the following dbt adapters, which have custom responses:
+
+        - https://github.com/dbt-labs/dbt-bigquery: BigQueryAdapterResponse
+        - https://github.com/databricks/dbt-databricks: DatabricksAdapterResponse
+        - https://github.com/dbt-labs/dbt-snowflake: SnowflakeAdapterResponse
+        - https://github.com/starburstdata/dbt-trino: TrinoAdapterResponse
+        """
+        # Remove the unnecessary metadata from the base AdapterResponse.
+        adapter_response.pop("_message", None)
```

Comment:
If there isn't a huge number of distinct fields here that we care about, I would prefer an implementation that explicity whitelists the fields that we want to capture rather than excludes the ones we don't, simply to avoid any nasty situations where some huge blob of metadata suddenly gets added to these responses and filling up the database (e.g. maybe the Trino AdapterResponse suddenly starts emitting the full query as "full_query" instead of "query").

Alternatively, some sort of size check on the field could be a reasonable safeguard

Reply:
Refactored to add an allowlist!